### DISCUSSION TITLE
changedetection Playwright Chromium/Javascript not working

### DISCUSSION THREAD

**Discussion by Lockie85** [tteck/Proxmox#3822] [Q&A]
Hey,

I've run the changedetection script which looked to install fine but when trying to use the Playwright Chromium/Javascript function within changedetection I appear to get this error:

`Exception: BrowserType.connect_over_cdp: WebSocket error: getaddrinfo ENOTFOUND browserless-chrome Call log: <ws connecting> ws://browserless-chrome:3000/chrome - <ws error> ws://browserless-chrome:3000/chrome error getaddrinfo ENOTFOUND browserless-chrome - <ws connect error> ws://browserless-chrome:3000/chrome getaddrinfo ENOTFOUND browserless-chrome - <ws disconnected> ws://browserless-chrome:3000/chrome code=1006 reason=`

Are there any checks I can run to see what might be wrong?

Many Thanks

**Comment by Lockie85** [comment]
@tteck sorry to ping you but I wonder if you might be able to help with this issue? Many thanks

**Reply by tteck** [reply]
I don't use Change Detection, but if you use the [search feature](https://github.com/tteck/Proxmox/discussions?discussions_q=changedetection) you _may_ find your answer.

**Reply by Lockie85** [reply]
The error I'm seeing with systemctl status browserless.service is:

```
Sep 30 10:10:34 changedetection browserless[148]:   Installed Browsers:
Sep 30 10:10:34 changedetection browserless[148]:       at file:///opt/browserless/build/browserless.js:203:23
Sep 30 10:10:34 changedetection browserless[148]:       at Array.forEach (<anonymous>)
Sep 30 10:10:34 changedetection browserless[148]:       at Browserless.start (file:///opt/browserless/build/browserless.js:198:19)
Sep 30 10:10:34 changedetection browserless[148]: } reason: Error: Couldn't load route "/chrome/content?(/)" due to missing browser binary for "ChromeCDP".
Sep 30 10:10:34 changedetection browserless[148]: Installed Browsers:
Sep 30 10:10:34 changedetection browserless[148]:     at file:///opt/browserless/build/browserless.js:203:23
Sep 30 10:10:34 changedetection browserless[148]:     at Array.forEach (<anonymous>)
Sep 30 10:10:34 changedetection browserless[148]:     at Browserless.start (file:///opt/browserless/build/browserless.js:198:19)
Sep 30 10:10:34 changedetection browserless[148]: 2024-09-30T09:10:34.128Z browserless.io:index.js:info  Process is finished, exiting
```

There doesn't appear to be any existing issues referencing this problem from my search.

**Reply by Lockie85** [reply]
Fixed it. It's something to do with either the paths or IPs. Chrome isn't the dir, it looks to have a version number now.

**Reply by paranoiak** [reply]
> Fixed it. It's something to do with either the paths or IPs. Chrome isn't the dir, it looks to have a version number now.

Hi, how did you solved it? Thanks

**Reply by bamzero** [reply]
> Fixed it. It's something to do with either the paths or IPs. Chrome isn't the dir, it looks to have a version number now.

Don't leave us hanging, any pointers on what to change? I've tried to work my way around it but haven't managed to get it running yet.

**Reply by bladel** [reply]
@Lockie85 how did you fix it?

**Reply by Lockie85** [reply]
No idea what I did. It's why I asked for help. I know I changed a few things simply by googling and trying things and it suddenly worked.

I've since deleted the LXC and tried to repro it and I can't.

**Reply by Lockie85** [reply]
Probably worth being aware of this though:
https://github.com/browserless/browserless/issues/4270#issuecomment-2403698003

**Reply by bamzero** [reply]
> Probably worth being aware of this though: [browserless/browserless#4270 (comment)](https://github.com/browserless/browserless/issues/4270#issuecomment-2403698003)

I saw that comment and I was a little confused. That version was released 4 months ago, it's currently v2.22.0-beta-2. The OP reported the problem while using v2.18

**Reply by YoSafeguard** [reply]
Guys, I also can’t overcome a similar problem! If anyone solves it please answer!

**Reply by Lockie85** [reply]
Issue looks to be resolved now in full for me when doing a fresh install.

**Reply by bamzero** [reply]
> Issue looks to be resolved now in full for me when doing a fresh install.

Can confirm, working here on fresh install also.

**Comment by Lockie85** [comment]
Anyone manage to change the width and height? Changing it in the file
/etc/systemd/system/changedetection.service

Doesn't seem to work for me.
